subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/git/tasks/CloneGitRepositoryTask.java

```diff
@@ -120,4 +128,8 @@ public int getDepth() {
     public void setDepth(int depth) {
         this.depth = depth;
     }
+
+    private boolean isTargetEmpty() {
+        return !targetDir.exists() || targetDir.list().length == 0;
```

Comment:
I'm curious if this is our long term strategy for clone task. Let's flesh out all the use cases and identify what should be the desired behavior of the clone task. For example:

- for given external repo, there should be 1 clone repo task for this repo
- if the outputs were tinkered with, we should clear the contents and clone the repo again
- if the outputs are the same, the task is UP-TO-DATE

Perhaps the problem with clone tasks that you have encountered is a sign that we should push harder on integrating test-downstream + upgrade-downstream plugin.